streaming/rdm_atomic.c

```diff
@@ -197,37 +300,74 @@ static int run_op(void)
 	count = (size_t *) malloc(sizeof(size_t));
 	ft_sync();
 
-
-	ft_start();
 	switch (op_type) {
 	case FI_MIN:
 	case FI_MAX:
+	case FI_SUM:
+	case FI_PROD:
+	case FI_LOR:
+	case FI_LAND:
+	case FI_BOR:
+	case FI_BAND:
+	case FI_LXOR:
+	case FI_BXOR:
 	case FI_ATOMIC_WRITE:
-		ret = is_valid_base_atomic_op(op_type);
-		if (ret > 0) {
-			for (i = 0; i < opts.iterations; i++) {
-				ret = execute_base_atomic_op(op_type);
-				if (ret)
-					break;
+		for (datatype = 0; datatype <= FI_LONG_DOUBLE_COMPLEX; datatype++) {
+			ft_start();
```

Comment:
There is no matching ft_stop() for each iteration of datatype. Times for ops that do both base and fetch atomic ops would be incorrect.